### PR TITLE
feat: Resolve pure variable values

### PR DESCRIPTION
**Pull request by polok**

### CODE REVIEW

File: Sources/FeaturevisorSDK/Instance.swift [hunk 1]
```diff
@@ -392,14 +392,15 @@ public class FeaturevisorInstance {
...
+          variableKey: VariableKey,
           context: Context = [:]
       ) ->  VariableValue? {
           return getVariable(featureKey: feature.key, variableKey: variableKey, context: context)
       }
+    
       func getVariable(
           featureKey: FeatureKey,
-          variableKey: String,
+          variableKey: VariableKey,
```
Comment (fahad19) [inline Sources/FeaturevisorSDK/Instance.swift:403]: 💖

File: Sources/FeaturevisorSDK/Instance.swift [hunk 2]
```diff
@@ -413,73 +414,61 @@ public class FeaturevisorInstance {
...
-    //    //return getValueByType(variableValue, "boolean") as boolean | undefined;
-    //  }
+      func getVariableBoolean(
+        feature: Feature,
+        variableKey: VariableKey,
+        context: Context) -> Bool? {
+            return getVariableBoolean(featureKey: feature.key, variableKey: variableKey, context: context)
+      }
+    
+      func getVariableBoolean(
```
Comment (fahad19) [inline Sources/FeaturevisorSDK/Instance.swift:424]: any reason why the method is repeating? I see `getVariableBoolean()` appear twice now.

same for `getVariableString()` few lines below.

File: Tests/FeaturevisorTypesTests/Extensions/Types+Equatable.swift [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        case (.not(let lhsNot), .not(let rhsNot)):
+            return lhsNot.not == rhsNot.not
+        default:
+            return false
+        }
+    }
+}
+
+extension VariableValue: Equatable {
+    public static func ==(lhs: VariableValue, rhs: VariableValue) -> Bool {
```
Comment (fahad19) [inline Tests/FeaturevisorTypesTests/Extensions/Types+Equatable.swift:78]: I can only imagine the pain you are going through for making it work in Swift 😬 

we take it soooo for granted in JS/TS world. thank you!

Comment (polok) [reply Tests/FeaturevisorTypesTests/Extensions/Types+Equatable.swift:78]: Hahahha, yeaaa but I'm almost there I think :D

File: Sources/FeaturevisorSDK/Instance.swift (hunk 2 above)
Comment (polok) [reply Sources/FeaturevisorSDK/Instance.swift:424]: I was uncommenting the code where we had these two methods with one difference in parameters - `featureKey` vs `feature`. If we don't want to (I didn't check the JS SDK) then I can remove on of them here and the similar below ones.

Comment (fahad19) [reply Sources/FeaturevisorSDK/Instance.swift:424]: ahhh I notice it now! `feature` and `featureKey`.

given this is public API, my expectation is people will always pass the `featureKey`. 

the `feature` object is only possible to use internally in the SDK.

keeping that in mind, I would request we keep only `featureKey` similar to how it's done in JS SDK.

JS API: 

- Docs: https://featurevisor.com/docs/sdks/#type-specific-methods
- Code: https://github.com/fahad19/featurevisor/blob/8bb0ab21bca434ff9c02029894795ea09541b7aa/packages/sdk/src/instance.ts#L1176

Comment (polok) [reply Sources/FeaturevisorSDK/Instance.swift:394]: Pushed the update. The test results are the same ;)

**Review by fahad19 — APPROVED** [review]

**Pull request merged**